### ISSUE TITLE
CoercionError error returns an empty path and stops further type checks

### ISSUE THREAD

**Issue by majkelcc** [rmosolgo/graphql-ruby#2166]
When raising a `GraphQL::CoercionError` in a custom type I'm seeing two issues:

1. it stops further type checks from being executed
2. result error contains an empty `path` array

You can see the problem on the following example:

## Schema
```ruby
class ExampleSchema < GraphQL::Schema

  class CustomType < GraphQL::Schema::Scalar
    def self.coerce_input(value, ctx)
      raise GraphQL::CoercionError, "#{value} coercion error!"
    end
  end

  class ChangelogType < GraphQL::Schema::Object
    field :title, String, null: false
  end

  class CreateChangelogInput < GraphQL::Schema::InputObject
    argument :title, CustomType, required: true
    argument :content, CustomType, required: true
  end

  class Mutation < GraphQL::Schema::Object
    field :create_changelog, ChangelogType, null: false do
      argument :changelog, CreateChangelogInput, required: true
    end

    def create_changelog(changelog:)
    end
  end

  mutation(Mutation)
  error_bubbling(false)

end
```

## Query

```
mutation($changelog: CreateChangelogInput!) {
  createChangelog(changelog: $changelog) {
    title
  }
}

# variables
{
  "changelog": {
    "title": "title",
    "content": "content"
  }
}
```

## Result

```
{
  "errors": [
    {
      "message": "Variable changelog of type CreateChangelogInput! was provided invalid value",
      "locations": [
        {
          "line": 1,
          "column": 10
        }
      ],
      "extensions": {
        "value": {
          "title": "title",
          "content": "content"
        },
        "problems": [
          {
            "path": [],
            "explanation": "title coercion error!"
          }
        ]
      }
    }
  ]
}
```

## Issues with the result

1. `problems` key contains only the first type error
2. `path` contains an empty array

**Comment by majkelcc** [comment]
This is version `1.9.3`.

**Comment by rmosolgo** [comment]
Oh, so you expect it to contain _two_ errors, one for `title` and the other for `content`, is that correct?

**Comment by majkelcc** [comment]
@rmosolgo correct!

And in addition to that, I would expect the `path` to contain the path to the input key that caused that specific issue, like `[mutation, createChangelog, changelog, title]`.

**Comment by majkelcc** [comment]
Errors are hard 😬

**Comment by moger777** [comment]
I'm assuming a good workaround would be to create a custom scalar type on frontend. Does mean some duplication between front/backend but also means you can skip round trip to server to get the errors.

**Comment by moger777** [comment]
Never mind, according to https://www.apollographql.com/docs/graphql-tools/scalars/ clients don't support custom scalars.

**Comment by moger777** [comment]
I'm able to get path if I return nil instead of raising coerce value, of course, that messes up the custom error messaging so not ideal.

**Comment by coding-chimp** [comment]
You actually get the `path` if the `coerce_input` method of your custom scalar returns `nil` instead of raising a `GraphQL::CoercionError`. The built in scalars are doing the same.

I'm not sure if there is a downside to doing this, though. 🤔

**Comment by rmosolgo** [comment]
Thanks for sharing your work-around @coding-chimp. That gave me a clue at how to go about a fix. The only downside to returning `nil` is that you don't get to customize the error message. 

I've started a fix in #2796
